x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_list_table.tsx

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import type { Query, TimeRange } from '@kbn/es-query';
+import React from 'react';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import { InfraClientStartDeps } from '../../../../types';
+
+const getLensHostsTable = (
```

Comment:
I'm guessing this isn't really to be reviewed because it's generated but I think we'll need to document how to re-generate this when needed.
This is why I'd wish we could use a Lens-as-Saved-Object instead and just edit it in the Lens UI.

Reply:
If we decide to do it this way, I think all these formulas and fields for calculating the metrics will be stored somewhere, like the inventory model metrics.  So we would be generating the lens attributes based on these. Similar to how we do with TSVB https://github.com/elastic/kibana/tree/main/x-pack/plugins/infra/common/inventory_models/host/metrics/tsvb.  If you want to view it in Lens to edit it there you can take the generated attributes and paste it into the Lens Playground, save it, and start editing it in the Lens UI.  
<img width="1167" alt="Screen Shot 2022-08-24 at 8 18 24 AM" src="https://user-images.githubusercontent.com/1676003/186416467-49f4a20a-2a9b-45da-a517-466342310b1b.png">
See https://github.com/elastic/kibana/tree/main/x-pack/plugins/lens#getting-started

The saved object might be the better way to go, but I thought this might be easier then having to ship them with packages or metricbeat and worrying about the user editing them, but I need to check the status on read only Kibana assets.  Also being able to store the fields and formulas for possibly reusing.

Reply:
Yeah, if we can write some (strongly typed) code to generate these attributes and combine them then that would feel a lot better!

Reply:
What would storing this configuration as a saved object enable ? Sounds like we want this config to be readonly for customer, and if we don't have to update it at runtime neither it would be an unnecessary network call